src/workspace.rs

```diff
@@ -411,3 +429,170 @@ impl Workspace {
         f(root_manifest_path)
     }
 }
+
+/// Subset of cargo profile settings to configure defaults for building contracts
+pub struct Profile {
+    opt_level: OptLevel,
+    lto: Lto,
+    // `None` means use rustc default.
+    codegen_units: Option<u32>,
+    overflow_checks: bool,
+    panic: PanicStrategy,
+}
+
+impl Default for Profile {
+    fn default() -> Profile {
+        Profile {
+            opt_level: OptLevel::One,
+            lto: Lto::Bool(false),
+            codegen_units: None,
+            overflow_checks: false,
+            panic: PanicStrategy::Unwind,
+        }
+    }
+}
+
+impl Profile {
+    /// The preferred set of defaults for compiling a release build of a contract
+    pub fn default_contract_release() -> Profile {
+        Profile {
+            opt_level: OptLevel::Z,
+            lto: Lto::Bool(true),
+            codegen_units: Some(1),
+            overflow_checks: true,
+            panic: PanicStrategy::Abort,
+        }
+    }
+
+    /// Set any unset profile settings from the config.
+    ///
+    /// Therefore:
+    ///   - If the user has explicitly defined a profile setting, it will not be overwritten.
+    ///   - If a profile setting is not defined, the value from this profile instance will be added
+    fn merge(&self, profile: &mut value::Table) {
+        let mut set_value_if_vacant = |key: &'static str, value: value::Value| {
+            if !profile.contains_key(key) {
+                profile.insert(key.into(), value);
+            }
+        };
+        set_value_if_vacant("opt-level", self.opt_level.to_toml_value());
+        set_value_if_vacant("lto", self.lto.to_toml_value());
+        if let Some(codegen_units) = self.codegen_units {
+            set_value_if_vacant("codegen-units", codegen_units.into());
+        }
+        set_value_if_vacant("overflow-checks", self.overflow_checks.into());
+        set_value_if_vacant("panic", self.panic.to_toml_value());
+    }
+}
+
+/// The [`opt-level`](https://doc.rust-lang.org/cargo/reference/profiles.html#opt-level) setting
+#[derive(Clone, Copy, PartialEq, Eq, Debug, Hash, PartialOrd, Ord)]
+#[allow(unused)]
+pub enum OptLevel {
+    Zero,
```

Comment:
rename to `None` or `NoOptimizations`?

Reply:
We could also wrap `OptLevel` into an `Option` where `None` is simply that no optimizations are performed.